application/client.core/src/app/environment/components/sidebar/shell/environment/component.ts

```diff
@@ -1,198 +1,125 @@
-            this._changeFocus();
-        }
-        if (this._ng_selected !== undefined) {
-            this._ng_selected.selected = false;
-        }
-        this._ng_selected = env;
-        this._ng_selected.selected = true;
-    }
-
-    public _ng_onDoubleClick(env: IEnvironment, type: string) {
-        this._changeFocus(env);
-        if (type === this._ng_value) {
-            env.editing.value = true;
-            this._prevValue = env.value;
-        } else {
-            env.editing.variable = true;
-            this._prevVariable = env.variable;
-        }
-    }
-
-    public _ng_add() {
-        this._ng_selected = undefined;
-        this._ng_adding = true;
-        if (this._ng_variableAdd !== undefined) {
-            this._ng_variableAdd.nativeElement.scrollIntoView();
-        }
-    }
-
-    public _ng_remove() {
-        if (this._ng_selected !== undefined) {
-            this.information.env = this.information.env.filter((env: IEnvironment) => {
-                return (env.value !== this._ng_selected.value && env.variable !== this._ng_selected.variable);
-            });
-        }
-    }
-
-    public _ng_onCancel() {
-        this._finishCreating(false);
-        this.close();
-    }
-
-    private _changeFocus(newEditEnv?: IEnvironment) {
-        if (this._editing !== undefined) {
-            this._dropEditChanges();
-            this._editing.editing.value = false;
-            this._editing.editing.variable = false;
-        }
-        this._finishCreating(false);
-        this._editing = newEditEnv;
-    }
-
-    private _dropEditChanges() {
-        if (this._editing.editing.value) {
-            this._editing.value = this._prevValue;
-        } else if (this._editing.editing.variable) {
-            this._editing.variable = this._prevVariable;
-        }
+    public _ng_onSetPwd() {
+        this.service.setPwd({ session: this._sessionID }).then((value: string) => {
+            if (value.trim() !== '') {
+                this.service.selectedPreset.information.pwd = value;
+                this.service.setPreset(this._sessionID, { pwd: value }).catch((error: Error) => {
+                    this._logger.error(error);
+                });
+            }
+        }).catch((error: Error) => {
+            this._logger.error(error);
+        });
     }
 
-    private _finishCreating(save: boolean) {
-        if (!this._ng_adding) {
-            return;
-        }
-        if (save) {
-            this.information.env.push({
-                custom: true,
-                editing: {
-                    value: false,
-                    variable: false,
-                },
-                value: this._ng_newValue,
-                variable: this._ng_newVariable,
-                selected: false,
+    public _ng_onShellChange(event: MatSelectChange) {
+        this.service.setEnv({ session: this._sessionID, shell: event.value }).then(() => {
+            this.service.selectedPreset.information.shell = event.value;
```

Comment:
Is it okay to have direct access to properties of service? It looks quite dangerous
```
this.service.selectedPreset.information.shell = event.value;
```
Maybe it makes sense to have something like
```
this.service.setShell(event.value);
```